tasks/main.yml

```diff
@@ -2,33 +2,36 @@
 - include: packages-{{ ansible_pkg_mgr }}.yml
 
 - name: Create Caddy user
-  user: name=caddy system=yes createhome=yes
+  user: name={{ caddy_user }} system=yes createhome=yes home={{ caddy_home }}
+
+- name: Create directories
+  file: path={{ item }} state=directory owner={{ caddy_user }} mode=0770
+  with_items:
+    - /etc/caddy
+    - /etc/ssl/caddy
+    - /var/www
+    - "{{ caddy_home }}"
+    - /var/log/caddy
 
 - name: Get all Caddy releases
-  get_url: url=https://api.github.com/repos/mholt/caddy/git/refs/tags dest=/home/caddy/releases.txt force=yes
+  get_url: url=https://api.github.com/repos/mholt/caddy/git/refs/tags dest={{ caddy_home }}/releases.txt force=yes
   when: caddy_update
   register: caddy_releases_cache
 
 - name: Set Caddy features
-  copy: content="{{caddy_features}}" dest=/home/caddy/features.txt
+  copy: content="{{caddy_features}}" dest={{ caddy_home }}/features.txt
   when: caddy_update
   register: caddy_features_cache
 
 - name: Download Caddy
-  get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{caddy_features}} dest=/home/caddy/caddy.tar.gz force=yes
+  get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{caddy_features}} dest={{ caddy_home }}/caddy.tar.gz force=yes
   when: caddy_releases_cache.changed or caddy_features_cache.changed
   register: caddy_binary_cache
 
-- name: Download Caddy
```

Comment:
Okay. I will remove this in a clean pull request.
How about the changes in the service files?

If you really want to copy them over, i can add a condition to vars with default from template. And let the user decide if they want the github&replace-way.